openrouteservice/src/main/java/heigit/ors/routing/RoutingProfile.java

```diff
@@ -853,12 +853,39 @@ else if (bearings[1] == null)
                 }
             }
 
-            if ((profileType == RoutingProfileType.DRIVING_HGV && HeavyVehicleAttributes.HGV == searchParams
-                    .getVehicleType()) && weightingMethod == WeightingMethod.RECOMMENDED) {
-                req.setWeighting("fastest");
-                req.getHints().put("weighting_method", "recommended_pref");
-
-                flexibleMode = true;
+            // MARQ24 for what ever reason after the 'weighting_method' hint have been set (based
+            // on the given searchParameter Max have decided that's necessary 'patch' the hint
+            // for certain profiles...
+            // ...and BTW if the flexibleMode set to true, CH will be disabled!
+            if(weightingMethod == WeightingMethod.FASTEST){
+                if (profileType == RoutingProfileType.CYCLING_TOUR) {
```

Comment:
so if the cycling tour profile is selected it will never use CH?

Reply:
correct - as long as "FASTEST" is used - the goal of my adjustment was to keep the original code (for what ever reason Max have added it) so that we can start with the release of 4.7...
Later I want to investigate "why" Max have build it this way - and either the complete stuff can be removed (I will hope so) - or we at least don't break things that habe been working in 4.5